lib/api/src/modules/url.ts

```diff
@@ -157,10 +153,43 @@ export const init: ModuleFn = ({ store, navigate, state, provider, fullAPI, ...r
   };
 
   const initModule = () => {
+    const initialArgs: Record<string, Story['args']> = {};
+
+    // Sets the args param to include any args that don't have their initial value
+    const updateArgsParam = ({ storyId, args }: { storyId: string; args?: Story['args'] }) => {
+      const customizedArgs = Object.entries(args || {}).reduce((acc, [key, value]) => {
+        if (!deepEqual(value, initialArgs[storyId][key])) acc[key] = value;
+        return acc;
+      }, {} as Story['args']);
+      const argsString = stringifyArgs(customizedArgs);
+      const argsParam = argsString.length ? `&args=${argsString}` : '';
+      queryNavigate(`${fullAPI.getUrlState().path}${argsParam}`, { replace: true });
+      api.setQueryParams({ args: argsString });
+    };
+
     fullAPI.on(NAVIGATE_URL, (url: string, options: { [k: string]: any }) => {
       fullAPI.navigateUrl(url, options);
     });
 
+    // Track initial args for the currently selected story
+    fullAPI.on(SET_STORIES, () => {
+      const data = fullAPI.getCurrentStoryData();
+      const storyArgs = isStory(data) ? data.args : undefined;
+      initialArgs[data.id] = storyArgs;
+    });
+
+    // Update URL with args for the selected story
+    // Set initial args if we haven't visited this story before
+    fullAPI.on(SET_CURRENT_STORY, () => {
+      const data = fullAPI.getCurrentStoryData();
+      const args = isStory(data) ? data.args : undefined;
+      initialArgs[data.id] = initialArgs[data.id] || args;
+      updateArgsParam({ storyId: data.id, args });
```

Comment:
Is this line necessary? By definition all the args are "initial", right?

Reply:
This restores any previously customized args when switching back and forth between stories. So no, they're not always initial.

Of course this assumes we want this behavior of restoring args to a previously customized state rather than resetting to the initial args. I think we do.